test/plugins/validation/oas3/pagination-ibm.test.js

```diff
@@ -147,6 +147,95 @@ describe('validation plugin - semantic - pagaination - oas3', function() {
     });
   });
 
+  it('should not complain when limit and offset params in allOf schema when other schemas provided', function() {
+    const spec = {
+      paths: {
+        '/resources': {
+          get: {
+            summary: 'this is a summary',
+            operationId: 'operationId',
+            parameters: [
+              {
+                name: 'limit',
+                in: 'query',
+                description: 'limit',
+                required: false,
+                schema: {
+                  type: 'integer',
+                  default: 10,
+                  maximum: 50
+                }
+              }
+            ],
+            responses: {
+              '200': {
+                content: {
+                  'application/json': {
+                    schema: {
+                      description: 'A collection of resources',
+                      type: 'object',
+                      properties: {
+                        resources: {
+                          description: 'An array of resources',
+                          type: 'array',
+                          items: {
+                            type: 'string'
+                          }
+                        }
+                      },
+                      allOf: [
+                        {
+                          description: 'Pagination response properties',
+                          type: 'object',
+                          required: ['limit', 'offset'],
+                          properties: {
+                            limit: {
+                              description:
+                                'The maximum number of items returned in this response',
+                              type: 'integer',
+                              format: 'int32'
+                            },
+                            offset: {
+                              description:
+                                'The offset in the collection of the first element in this page',
+                              type: 'integer',
+                              format: 'int32'
+                            }
+                          }
+                        },
+                        {
+                          description: 'Pagination response properties',
+                          type: 'object',
+                          required: ['prop1', 'prop2'],
+                          properties: {
+                            prop1: {
+                              description: 'prop1',
+                              type: 'integer',
+                              format: 'int32'
+                            },
+                            prop2: {
+                              description: 'prop2',
+                              type: 'integer',
+                              format: 'int32'
+                            }
+                          }
+                        }
```

Comment:
It doesn't seem like this second oneOf element is needed to verify the behavior this test is checking.  I'd recommend removing it.